src/Microsoft.OpenApi.OData.Reader/Common/EdmModelHelper.cs

```diff
@@ -61,6 +62,36 @@ internal static OpenApiSchema GetDerivedTypesReferenceSchema(IEdmStructuredType
             };
 
             return schema;
+        }            
+
+        /// <summary>
+        /// Verifies whether the provided navigation restrictions allow for navigability of a navigation property. 
+        /// </summary>
+        /// <param name="restrictionType">The <see cref="NavigationRestrictionsType"/>.</param>
+        /// <param name="restrictionProperty">The <see cref="NavigationPropertyRestriction"/>.</param>
+        /// <returns></returns>
+        internal static bool NavigationRestrictionsAllowsNavigability(
+            NavigationRestrictionsType restrictionType,
+            NavigationPropertyRestriction restrictionProperty)
+        {
+            // Verify using individual navigation restriction first
+            if (restrictionProperty != null && restrictionProperty.Navigability != null && restrictionProperty.Navigability.Value == NavigationType.None)
```

Comment:
```suggestion
            if (restrictionProperty?.Navigability?.Value == NavigationType.None)
```

Reply:
This suggestion throws the error:

![image](https://user-images.githubusercontent.com/40403681/155356081-457d3282-ad55-4b84-86cd-e19e55f9f734.png)

Perhaps?
`if (restrictionProperty?.Navigability != null && restrictionProperty.Navigability.Value == NavigationType.None)`

Reply:
yep, go with that, I thought the null propagation + implicit cast was smarter than that but apparently it's not.